test/DecompressTest.php

```diff
@@ -73,56 +101,73 @@ public function testCompressToFile()
     /**
      * Basic usage
      *
+     * @dataProvider returnFilterType
      * @return void
      */
-    public function testDecompressArchive()
+    public function testDecompressArchive($filterType)
     {
-        $filter  = new DecompressFilter('bz2');
-        $archive = $this->tmpDir . '/compressed.bz2';
+        $filter  = new DecompressFilter($filterType);
+        $archive = $this->tmpDir . '/compressed.' . $filterType;
         $filter->setArchive($archive);
 
         $content = $filter->compress('compress me');
         $this->assertTrue($content);
 
-        $filter2  = new DecompressFilter('bz2');
+        $filter2  = new DecompressFilter($filterType);
         $content2 = $filter2($archive);
         $this->assertEquals('compress me', $content2);
     }
 
-    public function testFilterMethodProxiesToDecompress()
+    /**
+     * @dataProvider returnFilterType
+     */
+    public function testFilterMethodProxiesToDecompress($filterType)
     {
-        $filter  = new DecompressFilter('bz2');
-        $archive = $this->tmpDir . '/compressed.bz2';
+        $filter  = new DecompressFilter($filterType);
+        $archive = $this->tmpDir . '/compressed.' . $filterType;
         $filter->setArchive($archive);
 
         $content = $filter->compress('compress me');
         $this->assertTrue($content);
 
-        $filter2  = new DecompressFilter('bz2');
+        $filter2  = new DecompressFilter($filterType);
         $content2 = $filter2->filter($archive);
         $this->assertEquals('compress me', $content2);
     }
 
-    public function returnUnfilteredDataProvider()
+    public function returnUnfilteredDataProvider(): iterable
     {
-        return [
-            [null],
-            [new \stdClass()],
-            [[
-                'decompress me',
-                'decompress me too, please'
-            ]]
-        ];
+        foreach ($this->returnFilterType() as $parameter) {
+            yield [$parameter[0], new stdClass()];
```

Comment:
previously available `null` was missing here
```suggestion
            yield [null];
            yield [$parameter[0], new stdClass()];
```

Reply:
It went futher down into `testDecompressNullValueThrowsRuntimeException`. As that one is supposed to throw an exception, handling it in the test case above doesn't make sense to me.

Reply:
in the previous version the test was expecting `null` as return when `null` was provided as argument, so this PR is actually introducing a new behaviour.

Reply:
Ok, I see. It looks like `bzdecompress(null)` actually works:

```
driehle:/laminas-filter$ php7.3 -r "var_dump(bzdecompress(null));"
string(0) ""
driehle:/laminas-filter$ php7.4 -r "var_dump(bzdecompress(null));"
string(0) ""
driehle:/laminas-filter$ php8.0 -r "var_dump(bzdecompress(null));"
string(0) ""
driehle:/laminas-filter$ php8.1 -r "var_dump(bzdecompress(null));"
string(0) ""
```

That is different from `gzuncompress(null)`, which has never worked and which throws a type incompatibility error with PHP 8.1:

https://3v4l.org/VJfrk

```
Deprecated: gzuncompress(): Passing null to parameter #1 ($data) of type string is deprecated in /in/VJfrk on line 3

Warning: gzuncompress(): data error in /in/VJfrk on line 3
bool(false)

Fatal error: Uncaught Error: Call to undefined function bzdecompress() in /in/VJfrk:6
Stack trace:
#0 {main}
  thrown in /in/VJfrk on line 6
```

So we would have to keep accepting `decompress(null)` for the `Bz2` filter, but allow it with the `Gz` filter... :-\

Reply:
i would love to see all this fully typed and non-null... but yeah... need to keep accepting this.

Reply:
Well, there is an issue. Updating to laminas-coding-standard 2.0 has introduced `declare(strict_types=1)` in all files. This doesn't allow passing `null` to `bzuncompress()` anymore withouth getting a `TypeError`. I didn't recognize this at first, as my console commands posted above didn't include strict types. Here is the updated output:

```
driehle:/laminas-filter$ php7.3 -r "declare(strict_types=1);bzdecompress(null);"
PHP Fatal error:  Uncaught TypeError: bzdecompress() expects parameter 1 to be string, null given in Command line code:1
Stack trace:
#0 Command line code(1): bzdecompress(NULL)
#1 {main}
  thrown in Command line code on line 1
```

Output is the same for PHP 7.3, 7.4, 8.0 and 8.1.

I'll add something to cover null explicitly in the code.

Reply:
the declare strict only kicks in for files which define it. if a have a `test.php` w/o `declare(strict_types=1)` there wont be a type error.

https://www.php.net/manual/en/language.types.declarations.php#language.types.declarations.strict

> Strict typing applies to function calls made from within the file with strict typing enabled, not to the functions declared within that file. If a file without strict typing enabled makes a call to a function that was defined in a file with strict typing, the caller's preference (coercive typing) will be respected, and the value will be coerced.

Reply:
You are right, but strict types are declared in our tests as well. I still think that we should cover this in the source files, so that it works for both users having strict types enabled or not enabled. See 289743d for a proposed solution.